### PR TITLE
Add support for elasticsearch v2.0...v2.1

### PR DESCRIPTION
**Pull request by sergeygaychuk**
Tested on:
1. v1.7.3
2. v2.0
3. v2.1
    a) with delete-by-query plugin
    b) without delete-by-query plugin

Maybe someone can setup v2.x version on Travis CI.

### CODE REVIEW

File: spec/chewy/fields/time_fields_spec.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
   let(:range) { (time - 1.minute)..(time + 1.minute) }
 
   specify { expect(PostsIndex.total).to eq(3) }
-  specify { expect(PostsIndex.filter { published_at == o{range} }.count).to eq(2) }
-  specify { expect(PostsIndex.filter { published_at == o{range.min.utc..range.max.utc} }.count).to eq(2) }
-  specify { expect(PostsIndex.filter { published_at == o{[range.min.to_date..range.max.to_date]} }.count).to eq(1) }
```
Comment (sergeygaychuk) [inline spec/chewy/fields/time_fields_spec.rb]: I have some troubles with that line on v2.x. I'll open issue on elastic.

**Comment by pyromaniac** [comment]
Wow wow!

File: lib/chewy/query.rb [hunk 2]
```diff
@@ -992,7 +997,7 @@ def _response
           begin
             Chewy.client.search(_request)
           rescue Elasticsearch::Transport::Transport::Errors::NotFound => e
-            raise e if e.message !~ /IndexMissingException/
+            raise e if e.message !~ /IndexMissingException/ && e.message !~ /index_not_found_exception/
```
Comment (pyromaniac) [inline lib/chewy/query.rb]: Which exceptions can be here as well?

Comment (sergeygaychuk) [inline lib/chewy/query.rb]: If you look at [this PR](https://github.com/elastic/elasticsearch/pull/12023/files), you will see that IndexMissingException was changed to IndexNotFoundException.

Comment (pyromaniac) [inline lib/chewy/query.rb]: Then should not it be || here to support both of versions?

Comment (sergeygaychuk) [inline lib/chewy/query.rb]: It's !~ (not match).

For example: 
[3] pry(main)> '{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"products","index":"products"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"products","index":"products"},"status":404}'.!~(/IndexMissingException/)
=> true

'{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"products","index":"products"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"products","index":"products"},"status":404}'.!~(/index_not_found_exception/)
=> false

The same for strings which contains IndexMissingException

If we want to reraise exception, when string does not contain IndexMissingException and index_not_found_exception substrings, then we'll use next logic:
reraise exception if message doesn't contain IndexMissingException and index_not_found_exception.

Comment (pyromaniac) [inline lib/chewy/query.rb]: Ah you are right, sorry, I've mixed something

Comment (sergeygaychuk) [inline lib/chewy/query.rb]: Don't worry, I recheck it few times today :smile: Thanks

**Comment by pyromaniac** [comment]
So are you considering this PR as ready to be merged?

**Comment by sergeygaychuk** [comment]
I think yes. If I find some new issues with v2.x, then I'll do separate PR.

**Comment by sergeygaychuk** [comment]
@pyromaniac But we need to add some tests to Travis CI. Really,  I don't  how to do it :(

**Comment by pyromaniac** [comment]
I know :) Will do it eventually

**Comment by sergeygaychuk** [comment]
:) Thanks

**Pull request merged**